custom-aa/test/utils/utils.ts

```diff
@@ -0,0 +1,165 @@
+import * as hre from "hardhat";
+import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+
+import { EIP712Signer, Provider, types, utils, Wallet } from "zksync-web3";
+import { localConfig } from "../../../tests/testConfig";
+import * as ethers from "ethers";
+
+export const deployFactory = async (
+  privateKey: string = localConfig.privateKey,
+) => {
+  // Private key of the account used to deploy
+  const wallet = new Wallet(privateKey);
+  const deployer = new Deployer(hre, wallet);
+  const factoryArtifact = await deployer.loadArtifact("AAFactory");
+  const aaArtifact = await deployer.loadArtifact("TwoUserMultisig");
+
+  // Getting the bytecodeHash of the account
+  const bytecodeHash = utils.hashBytecode(aaArtifact.bytecode);
+
+  const factory = await deployer.deploy(
+    factoryArtifact,
+    [bytecodeHash],
+    undefined,
+    [
+      // Since the factory requires the code of the multisig to be available,
+      // we should pass it here as well.
+      aaArtifact.bytecode,
+    ],
+  );
+
+  console.log(`AA factory address: ${factory.address}`);
+
+  return factory;
+};
+
+export const deployMultisig = async (factoryAddress: string) => {
+  const AA_FACTORY_ADDRESS = factoryAddress;
+
+  const provider = new Provider(localConfig.L2Network);
+  // Private key of the account used to deploy
+  const wallet = new Wallet(localConfig.privateKey).connect(provider);
+  const factoryArtifact = await hre.artifacts.readArtifact("AAFactory");
+
+  const aaFactory = new ethers.Contract(
+    AA_FACTORY_ADDRESS,
+    factoryArtifact.abi,
+    wallet,
+  );
+
+  // The two owners of the multisig
+  const owner1 = Wallet.createRandom();
+  const owner2 = Wallet.createRandom();
+
+  // For the simplicity of the tutorial, we will use zero hash as salt
+  const salt = ethers.constants.HashZero;
+
+  // deploy account owned by owner1 & owner2
+  const tx = await aaFactory.deployAccount(
+    salt,
+    owner1.address,
+    owner2.address,
+    { gasLimit: 10000000 },
+  );
+  await tx.wait(1);
+
+  // Getting the address of the deployed contract account
+  const abiCoder = new ethers.utils.AbiCoder();
+  const multisigAddress = utils.create2Address(
+    AA_FACTORY_ADDRESS,
+    await aaFactory.aaBytecodeHash(),
+    salt,
+    abiCoder.encode(["address", "address"], [owner1.address, owner2.address]),
+  );
+  console.log(`Multisig account deployed on address ${multisigAddress}`);
+
+  console.log("Sending funds to multisig account");
+  // Send funds to the multisig account we just deployed
+  await (
+    await wallet.sendTransaction({
+      to: multisigAddress,
+      // You can increase the amount of ETH sent to the multisig
+      value: ethers.utils.parseEther("0.008"),
+    })
+  ).wait(2);
+
+  const multisigBalanceBefore = await provider.getBalance(multisigAddress);
```

Comment:
The function that includes this logic is called `deployMultisig` but here the function is doing more than just deploying the MultiSig; it is also funding it. I suggest refactoring this logic into another function